libs/common/parameters_utils.h

```diff
@@ -22,139 +22,31 @@
 
 PXR_NAMESPACE_USING_DIRECTIVE
 
-class PrimvarsRemapper
-{
-public:
-    PrimvarsRemapper() {}
-    virtual ~PrimvarsRemapper() {}  
-
-    virtual bool RemapValues(const UsdGeomPrimvar &primvar, const TfToken &interpolation, 
-        VtValue &value);
-    virtual bool RemapIndexes(const UsdGeomPrimvar &primvar, const TfToken &interpolation, 
-        std::vector<unsigned int> &indexes);
-    virtual void RemapPrimvar(TfToken &name, std::string &interpolation);
-};
-
-class InputAttribute {
-public:
+struct InputAttribute {
     InputAttribute() {}
-    
-    virtual const UsdAttribute* GetAttr() const { return nullptr; }
-    virtual bool Get(VtValue *value, double time) {return false;}
-    const SdfPath &GetConnection() const {return _connection;}
-    void SetConnection(const SdfPath& c) {_connection = c;}
-    virtual bool IsAnimated() const {return false;}
-    virtual void ValidatePrimPath(std::string &path) {};
-
-protected:
-    SdfPath _connection;
-};
-
-class InputValueAttribute : public InputAttribute {
-public:
-    InputValueAttribute(const VtValue& v, const SdfPath *c = nullptr) : 
-        _value(v),
-        InputAttribute()
-    {
-        if (c)
-            _connection = *c;
-    }
-    
-    bool Get(VtValue *value, double time) override
-    {
-        if (value) {
-            *value = _value;
-            return !_value.IsEmpty();
-        }
-        return false;
-    }
-    
-protected:
-    const VtValue& _value;
-};
-
-class InputUsdAttribute : public InputAttribute
-{
-public:
-    InputUsdAttribute(const UsdAttribute& attribute, SdfPath* connection = nullptr) : 
-        _attr(attribute), InputAttribute()
-    {
-        if (connection)
-            _connection = *connection;
-        else if (_attr && _attr.HasAuthoredConnections()) {
-            SdfPathVector c;
-            _attr.GetConnections(&c);
-            if (!c.empty())
-                _connection = c[0];
-        }
-    }
-    const UsdAttribute* GetAttr() const override { return &_attr; }
-    bool Get(VtValue *value, double time) override
-    {
-        if (!_attr || !value)
-            return false;
-        
-        return _attr.Get(value, time);
-    }
-    bool IsAnimated() const override {return _attr.ValueMightBeTimeVarying();}
-    void ValidatePrimPath(std::string &path) override;
+    ~InputAttribute() {delete timeValues;}
```

Comment:
we should check if timeValues is not nullptr before deleting it, specially as the default constructor doesn't allocate it

Reply:
`delete nullptr`  is supposed to be safe, isn't it ?

Reply:
it is, and timeValues is correctly initialised, my bad.